common/djangoapps/student/forms.py

```diff
@@ -47,41 +58,42 @@ def clean_email(self):
             raise forms.ValidationError(self.error_messages['unusable'])
         return email
 
-    def save(
-            self,
-            subject_template_name='emails/password_reset_subject.txt',
-            email_template_name='registration/password_reset_email.html',
-            use_https=False,
-            token_generator=default_token_generator,
-            from_email=configuration_helpers.get_value('email_from_address', settings.DEFAULT_FROM_EMAIL),
-            request=None
-    ):
+    def save(self, use_https=False, token_generator=default_token_generator, request=None, **kwargs):
         """
         Generates a one-use only link for resetting password and sends to the
         user.
         """
-        # This import is here because we are copying and modifying the .save from Django 1.4.5's
-        # django.contrib.auth.forms.PasswordResetForm directly, which has this import in this place.
-        from django.core.mail import send_mail
         for user in self.users_cache:
+            site = Site.objects.get_current()
+            message_context = get_base_template_context(site)
+
             site_name = configuration_helpers.get_value(
                 'SITE_NAME',
                 settings.SITE_NAME
             )
-            context = {
+
+            message_context.update({
                 'email': user.email,
                 'site_name': site_name,
-                'uid': int_to_base36(user.id),
-                'user': user,
-                'token': token_generator.make_token(user),
-                'protocol': 'https' if use_https else 'http',
-                'platform_name': configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
-            }
-            subject = loader.render_to_string(subject_template_name, context)
-            # Email subject *must not* contain newlines
-            subject = subject.replace('\n', '')
-            email = loader.render_to_string(email_template_name, context)
-            send_mail(subject, email, from_email, [user.email])
+                'request': request,
+                # TODO: This overrides what `get_base_template_context` proviedes, yet this make the tests passes
+                'platform_name': configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME),
+                'reset_link': '{protocol}://{site}{link}'.format(
+                    protocol='https' if use_https else 'http',
+                    site=site.domain,
```

Comment:
@nasthagiri Also this one (`site.domain`) could be revised, because I'm not completely sure it's correct.

Reply:
@bill-filler I don't believe it's a blocker for this PR, but can you or someone from your team help us define and write the best practices for getting site-specific configuration?  See this question plus the TODO above.

Reply:
@nasthagiri 
Good idea. I will add a story to our backlog to write a best practices for getting site-specific configuration